google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/ITRetryConformanceTest.java

```diff
@@ -0,0 +1,301 @@
+  }
+
+  @Test
+  public void test() throws Throwable {
+    Storage nonTestStorage = retryTestFixture.getNonTestStorage();
+    Storage testStorage = retryTestFixture.getTestStorage();
+
+    Ctx ctx = ctx(nonTestStorage, empty());
+
+    LOGGER.fine("Running setup...");
+    Ctx postSetupCtx =
+        mapping.getSetup().apply(ctx, testRetryConformance).leftMap(s -> testStorage);
+    LOGGER.fine("Running setup complete");
+
+    LOGGER.fine("Running test...");
+    Ctx postTestCtx =
+        REPLACE_STORAGE_IN_OBJECTS_FROM_CTX
+            .andThen(mapping.getTest())
+            .apply(postSetupCtx, testRetryConformance)
+            .leftMap(s -> nonTestStorage);
+    LOGGER.fine("Running test complete");
+
+    LOGGER.fine("Running teardown...");
+    REPLACE_STORAGE_IN_OBJECTS_FROM_CTX
+        .andThen(mapping.getTearDown())
+        .apply(postTestCtx, testRetryConformance);
+    LOGGER.fine("Running teardown complete");
+  }
+
+  /**
+   * Load all of the tests and return a {@code Collection<Object[]>} representing the set of tests.
+   * Each entry in the returned collection is the set of parameters to the constructor of this test
+   * class.
+   *
+   * <p>The results of this method will then be run by JUnit's Parameterized test runner
+   */
+  @Parameters(name = "{0}")
+  public static Collection<Object[]> testCases() throws IOException, NoSuchAlgorithmException {
+    RetryTests retryTests = loadRetryTestsDefinition();
+
+    List<RetryTest> retryTestCases =
+        retryTests.getRetryTestsList().stream()
+            .sorted(Comparator.comparingInt(RetryTest::getId))
+            .collect(Collectors.toList());
+
+    RpcMethodMappings rpcMethodMappings = new RpcMethodMappings();
+    List<Object[]> testCases = generateTestCases(rpcMethodMappings, retryTestCases);
+
+    Random rand = getRand();
+    Collections.shuffle(testCases, rand);
+
+    validateGeneratedTestCases(rpcMethodMappings, testCases);
+
+    return testCases;
+  }
+
+  private static RetryTests loadRetryTestsDefinition() throws IOException {
+    ClassLoader cl = Thread.currentThread().getContextClassLoader();
+
+    String testDataJsonResource = "com/google/cloud/conformance/storage/v1/retry_tests.json";
+    InputStream dataJson = cl.getResourceAsStream(testDataJsonResource);
+    assertNotNull(
+        String.format("Unable to load test definition: %s", testDataJsonResource), dataJson);
+
+    InputStreamReader reader = new InputStreamReader(dataJson, Charsets.UTF_8);
+    RetryTests.Builder testBuilder = RetryTests.newBuilder();
+    JsonFormat.parser().merge(reader, testBuilder);
+    return testBuilder.build();
+  }
+
+  private static List<Object[]> generateTestCases(
+      RpcMethodMappings rpcMethodMappings, List<RetryTest> testCases) {
+    String host = TEST_BENCH.getBaseUri().replaceAll("https?://", "");
+
+    List<Object[]> data = new ArrayList<>(testCases.size());
```

Comment:
nitpick: Why this initial capacity? Currently, testCases.size() is 5 (always_idempotent, conditionally_idempotent_retries_when_precondition_is_present, conditionally_idempotent_no_retries_when_precondition_is_absent, non_idempotent, non-retryable errors), but the actual length of this list will end up much bigger, i.e. 737 with the current mappings

Reply:
Left over, I've removed the explicit sizing.